RecoMET/METFilters/plugins/EcalBadCalibFilter.cc

```diff
@@ -0,0 +1,171 @@
+// -*- C++ -*-
+//
+// Package:    ecalBadCalibSep2017ListFilter
+// Class:      ecalBadCalibSep2017ListFilter
+//
+/**\class ecalBadCalibSep2017ListFilter ecalBadCalibSep2017ListFilter.cc
+ 
+ Description: <one line class summary>
+ Event filtering to remove events with anomalous energy intercalibrations in specific ECAL channels
+*/
+//
+// Original Authors:  D. Petyt
+//
+ 
+ 
+// include files
+ 
+#include <iostream>
+ 
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/global/EDFilter.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/ESHandle.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+ 
+#include "DataFormats/EcalRecHit/interface/EcalRecHitCollections.h"
+#include "DataFormats/DetId/interface/DetId.h"
+#include "DataFormats/EcalDetId/interface/EEDetId.h"
+ 
+#include "Geometry/CaloTopology/interface/CaloTopology.h"
+#include "Geometry/CaloEventSetup/interface/CaloTopologyRecord.h"
+ 
+#include "Geometry/CaloGeometry/interface/CaloGeometry.h"
+ 
+using namespace std;
+ 
+ 
+class ecalBadCalibSep2017ListFilter : public edm::global::EDFilter<> {
+ 
+public:
+ 
+  explicit ecalBadCalibSep2017ListFilter(const edm::ParameterSet & iConfig);
+  ~ecalBadCalibSep2017ListFilter() override {}
+ 
+private:
+ 
+  // main filter function
+ 
+ bool filter(edm::StreamID, edm::Event & iEvent, const edm::EventSetup & iSetup) const override; 
+ 
+  // input parameters
+ 
+  // eb rechit collection (from AOD)
+  const edm::EDGetTokenT<EcalRecHitCollection>  ebRHSrcToken_;  
+  // ee rechit collection (from AOD)
+  const edm::EDGetTokenT<EcalRecHitCollection>  eeRHSrcToken_;
+
+  //config parameters (defining the cuts on the bad SCs)
+  const double ebMin_;              // EB rechit et threshold
+  const double eeMin_;              // EE rechit et threshold
+ 
+  const std::vector<unsigned int> baddetEB_;    // DetIds of bad EB channels  
+  const std::vector<unsigned int> baddetEE_;    // DetIds of bad EE channels
+ 
+  const bool taggingMode_;
+  const bool debug_;                // prints out debug info if set to true
+ 
+};
+ 
+// read the parameters from the config file
+ecalBadCalibSep2017ListFilter::ecalBadCalibSep2017ListFilter(const edm::ParameterSet & iConfig)
+  : ebRHSrcToken_     (consumes<EcalRecHitCollection>(iConfig.getParameter<edm::InputTag>("EBRecHitSource")))
+  , eeRHSrcToken_     (consumes<EcalRecHitCollection>(iConfig.getParameter<edm::InputTag>("EERecHitSource")))
```

Comment:
consumes for rechits can be conditional on content of the baddet vectors: e.g. now baddetEB is empty. In this case there is no need to require consume of EBRecHitSource